File: api/DTOs/Menu.cs
using api.Models;

namespace api.DTOs;

public record MenuRecommendRequest(
    string Query,
    int TopN = 20
);

public record MenuCandidateDto(
    string Id,
    string CategoryNameFa,
    string PersianName,
    string EnglishName,
    string Ingredients,
    string? ImageUrl,
    int? PriceValue,
    List<MenuSizePrice>? Sizes,
    double Score
);

public record MenuRecommendResponse(
    string Query,
    int TopN,
    List<MenuCandidateDto> Candidates
);

File: client/src/app/models/menu.ts
export class MenuRecommnedReq
{
    query: string = '';
    topN: number = 10;
    topK: number = 3;
}

export interface MenuRecommedRes
{
    query: string;
    topN: number;
    messageFa: string;
    candidates: MenuCandiate[];
}

export interface MenuCandiate
{
    id: string;
    catgeryNameFa: string;
    persianName: string;
    englishName: string;
    ingredients: string;
    imageUrl: string;
    priceValue: number;
    sizes: MenuPriceSize[];
    score: number;
}

export interface MenuPriceSize
{
    size: string;
    price: string;
    priceValue: number;
}